token/cli/src/main.rs

```diff
@@ -3714,4 +3715,158 @@ mod tests {
         assert_eq!(ui_account.mint, token.to_string());
         assert_eq!(ui_account.owner, aux_string);
     }
+
+    #[test]
+    fn transfer_with_account_delegate() {
+        let (test_validator, payer) = validator_for_test();
+        let config = test_config(&test_validator, &payer, &spl_token::id());
+
+        let token = create_token(&config, &payer);
+        let source = create_associated_account(&config, &payer, token);
+        let destination = create_auxiliary_account(&config, &payer, token);
+        let delegate = Keypair::new();
+
+        let file = NamedTempFile::new().unwrap();
+        write_keypair_file(&delegate, &file).unwrap();
+
+        let ui_amount = 100.0;
+        mint_tokens(&config, &payer, token, ui_amount, source);
+
+        let ui_account = config
+            .rpc_client
+            .get_token_account(&source)
+            .unwrap()
+            .unwrap();
+        assert_eq!(ui_account.token_amount.amount, "100");
+        assert_eq!(ui_account.delegate, None);
+        assert_eq!(ui_account.delegated_amount, None);
+        let ui_account = config
+            .rpc_client
+            .get_token_account(&destination)
+            .unwrap()
+            .unwrap();
+        assert_eq!(ui_account.token_amount.amount, "0");
+
+        process_test_command(
+            &config,
+            &payer,
+            &[
+                "spl-token",
+                CommandName::Approve.into(),
+                &source.to_string(),
+                "10",
+                &delegate.pubkey().to_string(),
+            ],
+        ).unwrap();
+
+        let ui_account = config
+            .rpc_client
+            .get_token_account(&source)
+            .unwrap()
+            .unwrap();
+        assert_eq!(ui_account.delegate.unwrap(), delegate.pubkey().to_string());
+        assert_eq!(ui_account.delegated_amount.unwrap().amount, "10");
+
+        let result = process_test_command(
+            &config,
+            &payer,
+            &[
+                "spl-token",
+                CommandName::Transfer.into(),
+                &token.to_string(),
+                "10",
+                &destination.to_string(),
+                "--fund-recipient",
+                "--allow-unfunded-recipient",
```

Comment:
nit: Since you already created the `destination` on line 3726, you shouldn't need these flags